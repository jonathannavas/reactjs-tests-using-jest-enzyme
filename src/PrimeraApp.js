import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ( { saludo, subtitulo } ) => {

    return (
        <>
            <h5>{ saludo }</h5>
            <p>{ subtitulo }</p>
        </>
    );
     
}   

/* Definir propiedades para los tipos de datos de las variables */
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
// Valores de las propiedades por defecto
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;
