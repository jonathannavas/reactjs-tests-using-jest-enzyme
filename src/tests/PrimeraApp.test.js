import React from 'react';

import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('test on PrimeraApp', () => {

    test('should to show <PrimeraApp />', () => {
        const mensaje = 'Hola mundo';
        const wrapper = shallow( <PrimeraApp saludo={ mensaje } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('should return subtitle send by props', () => {
        const mensaje = 'Hola mundo';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow( <PrimeraApp saludo={ mensaje } subtitulo={subtitulo} /> );
        
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe( subtitulo );
    });
    

});