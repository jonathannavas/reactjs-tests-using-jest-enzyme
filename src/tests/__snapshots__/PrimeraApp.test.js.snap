// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on PrimeraApp should to show <PrimeraApp /> 1`] = `
<Fragment>
  <h5>
    Hola mundo
  </h5>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
